File: core/core-api/src/main/java/com/coffee/gu/cancel/CancelEventListener.java
package com.coffee.gu.cancel;

import com.coffee.gu.CancelEvent;
import com.coffee.gu.order.Order;
import com.coffee.gu.order.OrderReader;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

@Component
public class CancelEventListener {

    private final CancelService cancelService;
    private final OrderReader orderReader;

    public CancelEventListener(CancelService cancelService, OrderReader orderReader) {
        this.cancelService = cancelService;
        this.orderReader = orderReader;
    }

    @EventListener
    public void handle(CancelEvent event) {
        Order order = orderReader.getByOrderKey(event.getOrderKey());
        cancelService.cancel(order, event);
    }
}

File: core/core-api/src/main/kotlin/com/coffee/gu/cancel/CancelEventListener.kt
package com.coffee.gu.cancel

import com.coffee.gu.CancelEvent
import com.coffee.gu.order.OrderReader
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class CancelEventListener(
    private val cancelService: CancelService,
    private val orderReader: OrderReader,
) {
    @EventListener
    fun handle(event: CancelEvent) {
        val order = orderReader.getByOrderKey(event.orderKey)
        cancelService.cancel(order, event)
    }
}
